state.go

```diff
@@ -156,7 +156,7 @@ type OverflowState struct {
 
 type OverflowArgument struct {
 	Value interface{}
-	Type  ast.Type
+	Type  sema.Type
```

Comment:
Could you briefly explain this change?

Reply:
the ast type is not usefull to us bu the sema type is. the api in flow expects the sema type and not the ast type basically.

Reply:
underflow as a library do not have the responsibility of creating the program and checker so it does not have a way to do the convertion itself.